src/ngx_http_lua_ssl.c

```diff
@@ -34,4 +38,504 @@ ngx_http_lua_ssl_init(ngx_log_t *log)
 }
 
 
+static size_t
+ngx_http_lua_ssl_get_error(u_long e, u_char *ssl_err,
+    size_t ssl_err_len, const char *default_err, size_t default_err_len)
```

Comment:
Just put some notes here:
OpenSSL stores errors in a queue. And it might generate multiple errors in a call.
Therefore,
1. You need to drain the error queue to get all errors.
2. Someone might only fetch the first error, and leave the others in the queue.

Current implementation of this function doesn't satisfy the deduction 1 and is vulnerable to the deduction 2.